playground/backend/cmd/server/controller_test.go

```diff
@@ -372,8 +369,67 @@ func TestPlaygroundController_GetRunOutput(t *testing.T) {
 				if !strings.EqualFold(got.Output, tt.want.Output) {
 					t.Errorf("GetRunOutput() got = %v, want %v", got.Output, tt.want.Output)
 				}
-				if !reflect.DeepEqual(got.CompilationStatus, tt.want.CompilationStatus) {
-					t.Errorf("GetRunOutput() got = %v, want %v", got.CompilationStatus, tt.want.CompilationStatus)
+			}
+		})
+	}
+}
+
+func TestPlaygroundController_GetRunError(t *testing.T) {
+	ctx := context.Background()
+	pipelineId := uuid.New()
+	runError := "MOCK_RUN_ERROR"
+	conn, err := grpc.DialContext(ctx, "bufnet", grpc.WithContextDialer(bufDialer), grpc.WithInsecure())
+	if err != nil {
+		t.Fatalf("Failed to dial bufnet: %v", err)
+	}
+	defer conn.Close()
+	client := pb.NewPlaygroundServiceClient(conn)
+
+	type args struct {
+		ctx  context.Context
+		info *pb.GetRunErrorRequest
+	}
+	tests := []struct {
+		name    string
+		prepare func()
+		args    args
+		want    *pb.GetRunErrorResponse
+		wantErr bool
+	}{
+		{
+			name:    "pipelineId doesn't exist",
+			prepare: func() {},
+			args: args{
+				ctx:  ctx,
+				info: &pb.GetRunErrorRequest{PipelineUuid: pipelineId.String()},
+			},
+			want:    nil,
+			wantErr: true,
+		},
+		{
+			name: "run error output exist",
+			prepare: func() {
+				_ = cacheService.SetValue(ctx, pipelineId, cache.RunError, runError)
+			},
+			args: args{
+				ctx:  ctx,
+				info: &pb.GetRunErrorRequest{PipelineUuid: pipelineId.String()},
+			},
+			want:    &pb.GetRunErrorResponse{Output: runError},
+			wantErr: false,
+		},
```

Comment:
Can we also check the situation when the `pipelineId` exists but there is no run error output?

Reply:
Added.